src/utils/dataset_utils.py

```diff
@@ -526,39 +497,106 @@ def parse_file(self, fp, file_name):
                 except:
                     print('Malformatted Line!')
                     continue
-                
+
+                # Cleaning documents if specified
+                if self.clean_docs:
+                    content = clean_input(content)
+                    memex_raw_content = get_posting_html(memex_raw_content, self.term)
+                    memex_url = parse_url(memex_url)
+
                 doc_text = ""
-                if 'extracted_text' in self.content_field:
-                    doc_text += content
-                if 'memex_raw_content' in self.content_field:
-                    doc_text = doc_text[:-2] + clean_html(memex_raw_content) + "\"\'"
-                if 'url' in self.content_field:
-                    doc_text = doc_text[:-2] + clean_url(url) + "\"\'"
-                if 'memex_url' in self.content_field:
-                    doc_text = doc_text[:-2] + clean_url(memex_url) + "\"\'"
```

Comment:
Where does this functionality live now?

Reply:
The 3 lines below. It's now a loop over the fields included in extracted_fields.